packages/app-store/giphy/api/get.ts

```diff
@@ -0,0 +1,60 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z, ZodError } from "zod";
+
+import prisma from "@calcom/prisma";
+
+import { GiphyManager } from "../lib";
+
+const getSchema = z.object({
+  url: z.string(),
+});
+
+const giphyUrlRegexp = /^https:\/\/media.giphy.com\/media\/(.*)\/giphy.gif/g;
+
+/**
+ * This is an example endpoint for an app, these will run under `/api/integrations/[...args]`
+ * @param req
+ * @param res
+ */
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const userId = req.session?.user?.id;
+  if (!userId) {
+    return res.status(401).json({ message: "You must be logged in to do this" });
+  }
+  try {
+    const { url } = req.body;
+    var matches = giphyUrlRegexp.exec(url);
+    if (!matches || matches.length < 2) {
+      return res.status(422).json({ message: "Giphy URL is invalid" });
+    }
+    const giphyId = matches[1];
+    const gifImageUrl = await GiphyManager.getGiphyById(giphyId);
+    return res.status(200).json({ image: gifImageUrl });
+  } catch (error: unknown) {
+    console.error({ error });
+    if (error instanceof Error) {
+      return res.status(500).json({ message: error.message });
```

Comment:
If we error because the url is invalid we should return a 400 error. A 500 indicates that something crashed on our side and that's not the case here.